api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/PixPdqMessageUtil.java

```diff
@@ -285,15 +272,29 @@ private void updatePID(PID pid, Patient patient) throws HL7Exception {
         // Update the pid segment with data in the patient
 
         // PID-3
-        pid.getPatientIdentifierList(0).getAssigningAuthority().getUniversalID().setValue(config.getShrPatientRoot());
-        pid.getPatientIdentifierList(0).getAssigningAuthority().getNamespaceID().setValue(config.getShrPatientRoot());
-        pid.getPatientIdentifierList(0).getAssigningAuthority().getUniversalIDType().setValue(config.getUniversalIdType());
-        pid.getPatientIdentifierList(0).getIDNumber().setValue(patient.getId().toString());
+//        pid.getPatientIdentifierList(0).getAssigningAuthority().getUniversalID().setValue(config.getShrPatientRoot());
+//        pid.getPatientIdentifierList(0).getAssigningAuthority().getNamespaceID().setValue(config.getShrPatientRoot());
+//        pid.getPatientIdentifierList(0).getAssigningAuthority().getUniversalIDType().setValue(config.getUniversalIdType());
+//        pid.getPatientIdentifierList(0).getIDNumber().setValue(patient.getId().toString());
+
+        // Other identifiers
+        List<String> uuidList = mpiProperties.getPixIdentifierTypeUuidList();
+        for (PatientIdentifier patIdentifier : patient.getIdentifiers()) {
+            if (uuidList.contains(patIdentifier.getIdentifierType().getUuid())) {
+                CX patientId = pid.getPatientIdentifierList(pid.getPatientIdentifierList().length);
```

Comment:
Yes it is correct. I based on the [link](http://hl7api.sourceforge.net/v231/apidocs/ca/uhn/hl7v2/model/v231/segment/PID.html#getPatientIdentifierList(int))